### DISCUSSION TITLE
biuld from geany source windows

### DISCUSSION THREAD

**Discussion by hupeng111** [geany/geany#3227] [General]
How to compile geany's source code to get a brand new IDE windows

**Comment by elextr** [comment]
See the [wiki](https://wiki.geany.org/howtos/win32/msys2)

**Reply by hupeng111** [reply]
> 
https://github.com/geany/geany 
 I would like to know if I can use this source code to compile a usable IDE?

**Reply by hupeng111** [reply]
> See the [wiki](https://wiki.geany.org/howtos/win32/msys2)

Instead of using the installer on the official website

**Reply by elextr** [reply]
Thats what the wiki article I posted explains.

**Reply by hupeng111** [reply]
> Thats what the wiki article I posted explains.
I'm new to it，What are the differences between the five shell command lines, To build under Windows is to always use mingw64？

**Reply by elextr** [reply]
To be honest, if you are new to development it would be better to use the installer, not try to compile, especially on Windows.  Geany is developed for Linux, so it uses Linux tools, and Linux libraries.  On Windows it needs adaptors between it and Windows, thats what msys is.

But msys is a foundation, its not automatically set up for the tools and libraries Geany needs, so they need to be installed, which is what the commands are for.  Its not a beginners operation, unless you can install msys, understand the command line, can exactly follow the instructions, and occasionally sort out things that go wrong it will be a frustrating time.

If you just want to use Geany to develop your own software, please use the installer.

**Reply by hupeng111** [reply]
First of all, thank you for your patience, but my job is to look for some possibility that geany can be compiled and run. Now I follow the wiki steps to successfully create a geany executable, and can compile and run the C language program normally. Now my task is to make it use the scope debugger and geoprj plugins as well as the geoany installation package. But I didn't expect a good way, do you have any suggestions? Thank you

**Reply by elextr** [reply]
plugins and installer are covered by the wiki I posted.

**Reply by hupeng111** [reply]
I built a geany-plugins-bundle based on the wiki and put it under c/geany. But I didn't find debugging tools or others in this folder, and I don't know how he associated with geany, can you tell me how to use this plugin?Thank you

**Reply by elextr** [reply]
You don't say what language you want to use.

Geany does not include compilers or debuggers, it supports 60 programming languages and 10 markup languages, and several languages have several tools, which would we include?  Almost everybody would be downloading stuff they didn't want.

The debugger plugins both list which debugger is supported, GDB only.  That means C/C++ only (and maybe Python, but never tried).  Msys should allow you to install GDB with pacman (the way you got all the other tools) and other compilers as well I think.

**Reply by eht16** [reply]
> I built a geany-plugins-bundle based on the wiki and put it under c/geany. But I didn't find 

And Geany and Geany-Plugins do work?
Unfortunately the information in the wiki are a bit outdated. I started working on rewriting this wiki page but didn't finish it yet.
